Source/Public/Add-VSTeamWorkItemType.ps1

```diff
@@ -0,0 +1,46 @@
+function Add-VSTeamWorkItemType {
+   [cmdletbinding(SupportsShouldProcess=$true)]
+   param(
+      [vsteam_lib.ProcessTemplateValidateAttribute()]
+      [ArgumentCompleter([vsteam_lib.ProcessTemplateCompleter])]
+      $ProcessTemplate = $env:TEAM_PROCESS,
+
+      [parameter(Mandatory = $true)]
+      [Alias('Name')]
+      [string] $WorkItemType,
+
+      [string] $Description,
+
+      [ArgumentCompleter([vsteam_lib.ColorCompleter])]
+      [vsteam_lib.ColorTransformToHexAttribute()]
+      $Color = '000000',
+
+      [vsteam_lib.IconTransformAttribute()]
+      [ArgumentCompleter([vsteam_lib.IconCompleter])]
+      [string]$Icon = 'icon_asterisk'
+   )
+
+   $url =  _getProcessTemplateUrl $ProcessTemplate
+   if ($url) {$url += "/workitemtypes?api-version=" + (_getApiVersion Processes)}
+   else      {Write-Warning "Could not convert '$ProcessTemplate' into a process template"; return}
+
+   $body  = @{
+      name  = $WorkItemType
+      color = $Color
+      icon  = $Icon
+   }
+   if ($Description)  {$body['description'] =$Description }
+
+   if ($PSCmdlet.ShouldProcess($ProcessTemplate, "Add workitem '$WorkItemType' to process template"))  {
+      $resp = _callapi -Url $url -method  Post  -ContentType "application/json" -body (ConvertTo-Json $body) -erroraction stop
+      if ($resp) {
```

Comment:
when should there be not a response? If there is none shouldn't there be an exception? In which cases do you expect an empty response and not throw an exception?

Reply:
One place is in testing, if we only want to see the call was made and don't check what is done with result. Again it's mostly a habit / pattern. If something throws an error some things (the AD module for example) write a message to error but don't throw a terminating exception. There's a comment repeated in multiple places about  
>  # I catch because using -ErrorAction Stop on the Invoke-RestMethod
What that does is put post api-call logic in the try block, and echo the error out in the catch. I just let any message bubble up where it occurs, and only run the api-call logic if there was something to run it _on_. That's simpler / easier to read, I think.

Reply:
I've taken that out. It needed a try-catch because if I try to add the same type to multiple Process templates, and it fails in the middle I don't want the pipeline to do some that were possible and leave other possible ones not done.